# 연 | 습 | 문 | 제

## Q1 다음은 Calculator 클래스이다.

In [67]:
class Calulator:
    def __init__(self):
        self.value = 0
    
    def add(self, val):
        self.value += val
    self.value += 3

NameError: name 'self' is not defined

## 위 클래스를 상속하는 UpgradeCalculator를 만들고 값을 뺄 수 있는 minus 메서드를 추가해 보자.  
## 즉 다음과 같이 동작하는 클래스를 만들어야 한다.

In [ ]:
cal = UpgradeCalculator()
cal.add(10)
cal.minus(7)

print(cal.value)

In [8]:
# 문제 답
class Upgradecalculator(Calulator):
    def minus(self, val):
        self.value -= val

cal = Upgradecalculator()
cal.add(10)
cal.minus(7)

print(cal.value)

3


## Q2 객체 변수 value가 100 이상의 값은 가질 수 없도록 제한하는 MaxLimitCalculator 클래스  
## 를 만들어 보자. 즉 다음과 같이 동작해야 한다.

In [ ]:
cal = MaxLimitCalculator()
cal.add(50)
cal.add(60)

print(cal.value)

## 단 반드시 다음과 같은 Calculator 클래스를 상속해서 만들어야한다.

In [6]:
class Calculator:
    def __init__(self):
        self.value = 0
        
    def add(self, val):
        self.value += val

In [12]:
# 문제 답
class MaxLimitCalculator(Calculator):
    def add(self, val):
        self.value += val
        if self.value > 100:
            self.value = 100
        
cal = MaxLimitCalculator()
cal.add(50)
cal.add(60)

print(cal.value)

100


## Q3 다음 결과를 예측해 보자.

In [13]:
# 1.
all([1, 2, abs(-3)-3])
# 예상 : False, 이유는 all은 하나의 값이라도 0이하의 값이 되면 안되는데 abs(-3)은 3이고 이것에 -3을 하면 0이 되서 False가 된다.
# 2.
chr(ord('a')) == 'a'
# 예상 : True, 이유는 ord는 문자를 아스키코드(숫자)로 바꾸는 함수이고 이걸 또다시 chr함수로 변환하면 결국 원래 알파벳 a가 나온다 그렇게 때문에 값이 같다.

# 확인
print(all([1, 2, abs(-3) - 3]))
print(chr(ord("a")) == "a")

False
True


## Q4 filter와 lambda를 사용하여 리스트[1,-2,3,-5,8,-3]에서 음수를 모두 제거해 보자.

In [16]:
result = list(filter(lambda a: a > 0, [1, -2, 3, -5, 8, -3]))
print(result)

[1, 3, 8]


## Q5 234라는 10진수의 16진수는 다음과 같이 구할 수 있다.

In [22]:
hex(234)

'0xea'

## 이번에는 반대로 16진수 문자열 0xea를 10진수로 변경해 보자.

In [21]:
int('0xea', 16)

234

## Q6 map과 lambda를 사용하여 [1, 2, 3, 4] 리스트의 각 요솟값에 3이 곱해진 리스트 [3, 6, 9, 12]를 만들어 보자.

In [24]:
list(map(lambda x: x*3, [1, 2, 3, 4]))

[3, 6, 9, 12]

## Q7 다음 리스트의 최댓값과 최솟값의 합을 구해 보자.

In [25]:
print(max([-8, 2, 7, 5, -3, 5, 0, 1]))
print(min([-8, 2, 7, 5, -3, 5, 0, 1]))

7
-8


## Q8 17 / 3의 결과는 다음과 같다.

In [26]:
17 / 3

5.666666666666667

## 위와 같은 결괏값 5.666666666666667을 소숫점 4자리까지만 반올림하여 표시해 보자.

In [27]:
round((17/3), 4)

5.6667

## Q9 다음과 같이 실행할 때 입력값을 모두 더하여 출력하는 스크립트(/Users/kwj903/Downloads/workspace  
## /education/my_educate/jump-to-python/exercise_04_python_core/myargv.py)를 작성해 보자.

In [ ]:
# cd exercise_04_python_core
# python myargv.py 1 2 3 4 5 6 7 8 9 10


Traceback (most recent call last):
  File "/Users/kwj903/Downloads/workspace/education/my_educate/jump-to-python/exercise_04_python_core/myargv.py", line 4, in <module>
    print(sum(a))
          ^^^^^^
TypeError: unsupported operand type(s) for +: 'int' and 'str'


In [ ]:
# 답
# myargv.py
import sys

a = sys.argv
a.pop(0)
a = map(lambda x: int(x), a)
print(sum(a))

## Q10 os 모듈을 사용하여 다음과 같이 동작하도록 코드를 작성해 보자.
1. /Users/kwj903/Downloads/workspace/education/my_educate/jump-to-python 디렉토리로 이동한다.
2. dir(맥기준 : ls) 명령을 실행하고 그 결과를 변수에 담는다.
3. dir(맥기준 : ls) 명령의 결과를 출력한다.

In [46]:
import os
os.chdir("/Users/kwj903/Downloads/workspace/education/my_educate/jump-to-python")

f = os.popen("ls")
print(f.read())

README.md
exercise_01_data_type
exercise_02_control_flow
exercise_03_input_output
exercise_04_python_core
main.py
pyproject.toml
uv.lock



## Q11 glob 모듈을 사용하여 /Users/kwj903/Downloads/workspace/education/my_educate/jump-to-python  
## 디렉터리의 파일 중 확장자가 .py인 파일만 출력하는 프로그램을 작성해 보자.

In [50]:
import glob

a = glob.glob("/Users/kwj903/Downloads/workspace/education/my_educate/jump-to-python/*.py")
print(a)

['/Users/kwj903/Downloads/workspace/education/my_educate/jump-to-python/main.py']


## Q12 time 모듈을 사용하여 현재 날짜와 시간을 다음과 같은 형식으로 출력해 보자.

In [52]:
# 2018/04/03 17:20:32
import time
print(time.strftime('%Y/%m/%d %H:%m:%S', time.localtime(time.time())))


2026/02/27 01:02:18


## Q13 random 모듈을 사용하여 로또 번호(1~45 사이의 숫자 6개)를 생성해 보자(단 중복된 숫자가 있으면 안 됨.)

In [ ]:
import random
result = []
numbers = [i for i in range(1, 46)]
def rotto(numbers):
    for i in range(6):
        number = random.choice(numbers)
        numbers.remove(number)
        # print(numbers)
        result.append(number)
    return result
    
rotto_num = rotto(numbers)
print("행운은 번호는?: ", *rotto_num)
        

행운은 번호는?:  20 43 33 42 30 23
